validator/server_arb/execution_run.go

```diff
@@ -79,6 +83,96 @@ func (e *executionRun) GetStepAt(position uint64) containers.PromiseInterface[*v
 	})
 }
 
+func (e *executionRun) GetMachineHashesWithStepSize(machineStartIndex, stepSize, requiredNumHashes uint64) containers.PromiseInterface[[]common.Hash] {
+	return stopwaiter.LaunchPromiseThread(e, func(ctx context.Context) ([]common.Hash, error) {
+		return e.machineHashesWithStepSize(ctx, machineStartIndex, stepSize, requiredNumHashes)
+	})
+}
+
+func (e *executionRun) machineHashesWithStepSize(
+	ctx context.Context,
+	machineStartIndex,
+	stepSize,
+	requiredNumHashes uint64,
+) ([]common.Hash, error) {
+	if stepSize == 0 {
+		return nil, fmt.Errorf("step size cannot be 0")
+	}
+	if requiredNumHashes == 0 {
+		return nil, fmt.Errorf("required number of hashes cannot be 0")
+	}
+	machine, err := e.cache.GetMachineAt(ctx, machineStartIndex)
```

Comment:
Maybe this is already happening elsewhere. But, how does the algorithm know which message/block number to be validating the machine against?

Reply:
An `executionRun` is initialized with a validation entry, which is for a specific message number. So this is already relevant to the message we want to validate against

Reply:
Thanks.